pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -557,8 +561,7 @@ static void validateUpsertAndDedupConfig(TableConfig tableConfig, Schema schema)
         "Upsert/Dedup table must use low-level streaming consumer type");
     // replica group is configured for routing
     Preconditions.checkState(tableConfig.getRoutingConfig() != null
-            && RoutingConfig.STRICT_REPLICA_GROUP_INSTANCE_SELECTOR_TYPE.equalsIgnoreCase(
-            tableConfig.getRoutingConfig().getInstanceSelectorType()),
+            && isRoutingStrategyAllowedForUpsert(tableConfig.getRoutingConfig()),
```

Comment:
FYI: I left the message for this Precondition as is (it still says `i.e. strictReplicaGroup`). Reason is that multi-stage routing strategy is in beta so we don't want to bubble it up to the users yet.